Makefile

```diff
@@ -0,0 +1,138 @@
+# -*- makefile -*-
+# lind-wasm-apps unified build (hardened, with lmbench build wrapper)
+
+SHELL := /bin/bash
+.ONESHELL:
+.SHELLFLAGS := -eu -o pipefail -c
+
+# -------- Paths ---------------------------------------------------------------
+LIND_WASM_ROOT ?= $(HOME)/lind-wasm
+BASE_SYSROOT   ?= $(LIND_WASM_ROOT)/src/glibc/sysroot
+
+APPS_ROOT      := $(CURDIR)
+APPS_BUILD     := $(APPS_ROOT)/build
+APPS_OVERLAY   := $(APPS_BUILD)/sysroot_overlay
+MERGED_SYSROOT := $(APPS_BUILD)/sysroot_merged
+APPS_BIN_DIR   := $(APPS_BUILD)/bin
+APPS_LIB_DIR   := $(APPS_BUILD)/lib
+
+TOOL_ENV       := $(APPS_BUILD)/.toolchain.env
+JOBS ?= $(shell nproc 2>/dev/null || getconf _NPROCESSORS_ONLN || echo 4)
+
+# -------- Phonies -------------------------------------------------------------
+.PHONY: all preflight dirs print-config libtirpc merge-sysroot stubs lmbench clean clean-all
+
+all: preflight libtirpc merge-sysroot stubs lmbench
+
+print-config:
+	@echo "LIND_WASM_ROOT=$(LIND_WASM_ROOT)"
+	@echo "BASE_SYSROOT=$(BASE_SYSROOT)"
+	@echo "APPS_OVERLAY=$(APPS_OVERLAY)"
+	@echo "MERGED_SYSROOT=$(MERGED_SYSROOT)"
+	@echo "APPS_BIN_DIR=$(APPS_BIN_DIR)"
+	@echo "APPS_LIB_DIR=$(APPS_LIB_DIR)"
+	@if [[ -r '$(TOOL_ENV)' ]]; then . '$(TOOL_ENV)'; \
+	  echo "CLANG=$$CLANG"; echo "AR=$$AR"; echo "RANLIB=$$RANLIB"; fi
+
+dirs:
+	mkdir -p \
+	  '$(APPS_OVERLAY)/usr/include' \
+	  '$(APPS_OVERLAY)/usr/lib/wasm32-wasi' \
+	  '$(APPS_OVERLAY)/lib/wasm32-wasi' \
+	  '$(MERGED_SYSROOT)/include' \
+	  '$(MERGED_SYSROOT)/include/wasm32-wasi' \
+	  '$(MERGED_SYSROOT)/lib/wasm32-wasi' \
+	  '$(MERGED_SYSROOT)/usr/lib/wasm32-wasi' \
+	  '$(APPS_BIN_DIR)' \
+	  '$(APPS_LIB_DIR)'
+
+preflight: dirs
```

Comment:
I understand why this is useful but it also takes up a lot of space for things that aren't really the core of whats happening in this file.

@m-hemmings is there a more standard way to do this?

Reply:
You could possibly do something like this:

```
@$(foreach v, \
		LIND_WASM_ROOT BASE_SYSROOT APPS_OVERLAY MERGED_SYSROOT \
		APPS_BIN_DIR APPS_LIB_DIR TOOL_ENV, \
		printf "%s=%s\n" $(v) "$($(v))";)
```

that at least is more compact and would be nice and easy to maintain